modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/workflow/TenantCreationExternalWorkflowExecutor.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.neethi.Policy;
+import org.apache.neethi.PolicyEngine;
+import org.apache.rampart.RampartMessageData;
+import org.wso2.carbon.appfactory.common.AppFactoryConstants;
+import org.wso2.carbon.appfactory.common.AppFactoryException;
+import org.wso2.carbon.appfactory.common.util.AppFactoryUtil;
+import org.wso2.carbon.appfactory.core.internal.ServiceHolder;
+import org.wso2.carbon.appfactory.core.workflow.dto.TenantCreationWorkflowDTO;
+import org.wso2.carbon.appfactory.core.workflow.dto.WorkflowDTO;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.FileNotFoundException;
+import java.text.MessageFormat;
+
+/**
+ * The executor class for tenant creation external execution
+ */
+public class TenantCreationExternalWorkflowExecutor implements WorkflowExecutor {
+
+    private static Log log = LogFactory.getLog(TenantCreationExternalWorkflowExecutor.class);
+
+    private static Policy loadPolicy(String xmlPath) throws FileNotFoundException, XMLStreamException {
+        StAXOMBuilder builder = new StAXOMBuilder(xmlPath);
+        return PolicyEngine.getPolicy(builder.getDocumentElement());
+    }
+
+    /**
+     * Execute tenant creation external workflowDTO process
+     *
+     * @param workflowDTO the tenant creation workflowDTO dto
+     * @throws AppFactoryException AppFactoryException exception throws if not success the execution
+     */
+    @Override public void execute(WorkflowDTO workflowDTO) throws AppFactoryException {
+
+        if (log.isDebugEnabled()) {
+            String message = "Executing tenant creation external workflow for the tenant domain : " + workflowDTO
+                    .getTenantDomain();
+            log.debug(message);
+        }
+        String EPR;
+
+        if (!(workflowDTO instanceof TenantCreationWorkflowDTO)) {
+            String message = "TenantCreationWorkflowDTO type is expected but unexpected type is passed to the execute"
+                    + " method for the tenant domain : " + workflowDTO.getTenantDomain();
+            throw new IllegalArgumentException(message);
+        }
+
+        TenantCreationWorkflowDTO tenantCreationWorkflow = (TenantCreationWorkflowDTO) workflowDTO;
+
+        try {
+            EPR = AppFactoryUtil.getAppfactoryConfiguration().getFirstProperty(WorkflowConstant.BPS_SERVER_URL)
+                    + "CreateTenant";
+            if (StringUtils.isBlank(EPR)) {
+                String message = "Can not read EPR value from the appfactory.xml configuration, the value is null or "
+                        + "empty for the tenant domain : " + workflowDTO.getTenantDomain();
+                throw new AppFactoryException(message);
```

Comment:
This is wrong. Here you are throwing an exception catching that in the catch block.